st2actions/st2actions/runners/mistral/utils.py

```diff
@@ -0,0 +1,139 @@
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import re
+
+import six
+import yaml
+
+from st2common import log as logging
+from st2common.models.system.common import ResourceReference
+from st2common.persistence.action import Action
+
+
+LOG = logging.getLogger(__name__)
+
+REGEX_ACTION = re.compile("^[\w\.]+[^=\s\"]*")
+REGEX_ACTION_PARAMS = re.compile("([\w]+)=(\"[^\"]*\"\s*|'[^']*'\s*|"
+                                 "\{[^}]*\}\s*|\[.*\]\s*|[\.,:\w\d\.]+)")
+
+
+def _parse_cmd_and_input(cmd_str):
+    cmd_matcher = REGEX_ACTION.search(cmd_str)
```

Comment:
I am a complete regex n00b so I have no clue what this is doing. Can you add some docs or comments in the method to describe.